rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -628,90 +627,91 @@ def _create_model_data(
         training: bool = True,
     ) -> RasaModelData:
         """Prepare data for training and create a RasaModelData object"""
+        from rasa.utils.tensorflow import model_data_utils
 
-        # TODO: simplify model data creation
-        #   convert training data into a list of attribute to features and reuse some
-        #   of the methods of TED (they most likely need to change a bit)
-
-        features = defaultdict(lambda: defaultdict(list))
-        label_ids = []
-
-        for example in training_data:
-            if label_attribute is None or example.get(label_attribute):
-                text_features = self._extract_features(example, TEXT)
-                for feature_key, feature_value in text_features.items():
-                    features[TEXT][feature_key].append(feature_value)
+        attributes_to_consider = [TEXT, label_attribute]
+        if self.component_config[ENTITY_RECOGNITION]:
+            attributes_to_consider.append(ENTITIES)
 
-            # only add features for intent labels during training
-            if training and example.get(label_attribute):
-                label_features = self._extract_features(example, label_attribute)
-                for feature_key, feature_value in label_features.items():
-                    features[LABEL][feature_key].append(feature_value)
+        if training:
+            # only use those training examples that have the label_attribute set
+            # during training
+            training_data = [
+                example for example in training_data if label_attribute in example.data
+            ]
 
-                if label_id_dict:
-                    label_ids.append(label_id_dict[example.get(label_attribute)])
+        features_for_examples = model_data_utils.convert_training_examples(
+            training_data,
+            attributes_to_consider,
+            entity_tag_specs=self._entity_tag_specs,
+            featurizers=self.component_config[FEATURIZERS],
+            bilou_tagging=self.component_config[BILOU_FLAG],
+        )
+        attribute_data, _ = model_data_utils.convert_to_data_format(
+            features_for_examples, consider_dialogue_dimension=False
+        )
 
-            # only add tag_ids during training
-            if training and self.component_config.get(ENTITY_RECOGNITION):
-                for tag_spec in self._entity_tag_specs:
-                    features[ENTITIES][tag_spec.tag_name].append(
-                        self._tag_ids_for_crf(example, tag_spec)
-                    )
+        # during training we store the tag ids for the different entity labels under
+        # the key ENTITIES. during prediction we do not have any tag ids.
+        if not training and ENTITIES in attribute_data:
+            del attribute_data[ENTITIES]
```

Comment:
does it make sense to move it to `model_data_utils.convert_to_data_format`?

Reply:
not needed if you just add `ENTITIES` as attribute during training.